### DISCUSSION TITLE
Typerror: Unknow file extension ".css"

### DISCUSSION THREAD

**Discussion by amirtbi** [vitest-dev/vitest#2677] [Q&A] [answered]
I use Vitest in my Vue.js 3 app. I also use Vite 3.2.3 + Vuetify 3.0.1.
When I run a test for a component. the following error appears:
`TypeError: Unknown file extension ".css" for /home/file/filename/project/node_modules/vuetify/lib/components/VAvatar/VAvatar.css`

How can handle this issue?

**Comment by Tauromachian** [comment]
I have the same issue. Any solutions yet?

**Comment by cintek** [comment]
Did you try add the following to the configuration for Vitest?

```
deps: {
  inline: ["vuetify"],
}
```

**Reply by travs15** [reply]
This worked for me.

**Reply by Rusinas** [reply]
Doesn't work for me :(

**Reply by sombriks** [reply]
i used that and ended up with another problem:

```
Error: [Vuetify] Could not find defaults instance
```

**Reply by Unkrass** [reply]
This works, but shows a deprecation warning:
```
Vitest "deps.inline" is deprecated. If you rely on vite-node directly, use "server.deps.inline" instead. Otherwise, consider using "deps.optimizer.web.include"
```

**Reply by nadr0** [reply]
Does not work for me.

**Reply by nizhm** [reply]
Work for me. But i am using another ui framework(Nutui). The name follows `node_modules/` should be add to deps. i add `@nutui` to deps because file path is ".../node_modules/@nutui/..."

**Comment by CodeGetters** [comment]
I have the similar problem

**Comment by PanDymitr** [comment, marked as answer]
```typescript
// vitest: ^0.34.4
// vuetify: ^3.3.3

export default defineConfig({
  test: {
    server: {
      deps: {
        inline: ['vuetify'],
      },
    },
  }
})
```

**Reply by LeBenLeBen** [reply]
I had to specify the following as well for it to work:

```js
test: {
  deps: {
    optimizer: {
      web: {
        include: ['vuetify'],
      },
    },
  },
},
```

**Reply by ashomurodov** [reply]
it's not working me. When i try build i have to this issue.

An internal error occurred.
[vite-ssg] Please report an issue, if none already exists: https://github.com/antfu/vite-ssg/issues
node:internal/errors:496
    ErrorCaptureStackTrace(err);
    ^

TypeError: Unknown file extension ".css" for /home/ayyub/Desktop/work/cabinet-settings/node_modules/.pnpm/vuetify@3.3.17_typescript@5.2.2_vite-plugin-vuetify@1.0.2_vue@3.3.4/node_modules/vuetify/lib/directives/ripple/VRipple.css

**Reply by nadr0** [reply]
Neither of these configurations worked for me as well.

**Comment by NikolaVujanic** [comment]
> i used that and ended up with another problem:
> 
> ```
> Error: [Vuetify] Could not find defaults instance
> ```

This solution worked for me. According to [docs](https://vuetifyjs.com/en/getting-started/unit-testing/#setup-vitest), you should include vuetify to each test something like this:

`import { vuetify } from '@/plugins/vuetify'`

and then use in **wrapper**:

`const wrapper = mount(TestComponent, {
      global: {
        plugins: [vuetify],
      },
 });`

**Reply by LeBenLeBen** [reply]
You can also register the plugin globally once, using the `config` object from `@vue/test-utils`, see https://test-utils.vuejs.org/api/#config-global

**Comment by AlekseyMalakhov** [comment]
For me the following works:
1) Add `vuetify` to your `vite.config` (or If you, as me, have separate `vitest.config` add it there)
```
test: {
    server: {
      deps: {
        inline: ['vuetify'],
      },
    },
  }
```
2) Export your `vuetify` instance from where you created it (e.g. `@/plugins/vuetify`). I created it in `main.ts` so it looks like this
```
...
export const vuetify = createVuetify()

const app = createApp(App)
app.use(createPinia())
app.use(router)
app.use(vuetify)
app.mount('#app')
...
```
3. Import it to your test file and register it in `config.global.plugins`
```
import { mount, config } from '@vue/test-utils'
import { vuetify } from '@/main'
config.global.plugins = [vuetify]

...//go tests
```

**Reply by Diegovalen47** [reply]
This works for me, Thank you!

**Comment by Scorpibear** [comment]
It could be issues with other dependencies. e.g. in my case it was the error:
`TypeError: Unknown file extension ".css" for <...>\node_modules\@vojtechlanka\vue-tags-input\dist\style.css`

and the working fix was to add to vitest.config.js in test: {} section the following lines:
```
server: {
  deps: {
    inline: ['@vojtechlanka/vue-tags-input']
  }
}
```

**Reply by httpete-broadcom** [reply]
After upgrading to the latest mui-x-datagrid version 8 I needed this.

**Comment by Rusinas** [comment]
Hey. I've been struggling with for a loooong time but here is the solution:
```
export default defineConfig({
  // ...the rest
  test: {
    pool: 'vmThreads', // <-- the thing
  },
});
```

It comes with a price though and you are likely going to have another issues, but at least they're more explicit. For example, after this I had to mock some functions from `@unhead/vue` - and that was it. I don't why it helps or how it works or what the caveats are - don't ask me, I am just happy to fix this stupid issue. Hopefully for good. 

`deps.optimizer.web.include: ['vuetify']` or any other mention of 'vuetify' doesn't make any difference, so don't bother.
